api/controllers/StatsController.js

```diff
@@ -141,14 +141,19 @@ module.exports = {
     try {
       const seasons = await sails.helpers.getSeasonsWithoutRankings();
       const [currentSeason] = seasons;
-      const allUsers = User.find({});
+      const allUsers = User.find({
+        select: ['id', 'username'],
+      });
```

Comment:
Previously, this grabbed all data from every user record. Now we're only taking their username + id. This is the crux of the fix, and it's rolled out in a few places that queried for users and for games

Reply:
Does it make sense to query for only users where there `updatedAt` is between the season start and end times?

Reply:
That’s a good idea, best case you limit the data as low as humanly possible. When @itsalaidbacklife and I talked about this originally I wanted to cache it in some way and statically serve it, only updating after a game is played with some sort of other delay to stop slamming the server.

I think it would make sense to use a custom endpoint in sails and then server the cached version until the delay is up then re-request it. If you wanted to collect all the data like we used to do it would be even more
Important.

Reply:
Reasonable ideal. The user table itself doesn't update when games are completed so that doesn't actually get us the users who have played this season. Alternatively, we could wait for the query for games first and then get the list of user ids from the games, then query for the list of users based on those ids. That should theoretically use less memory, but take more time (because we need to query for games before we can get the users.

From what I can tell, updating the select statement to use less data per user record drastically reduces the memory footprint such that we can afford to query both simultaneously in order to load the page faster

Reply:
Does the user table get updated on `clearGame` ?

Reply:
That's an interesting point. I think that would be more efficient for the current setup, actually. My two concerns are that it would only work for the current season data, not an old one (since `updatedAt` will be overwritten compared the next time you play), and that this wouldn't work once we rework things so that the `user` table doesn't have a fk to game any more (just using the fk from game to user).

All in all I think that would be more efficient for right now, but maybe not as clean in the longer term

Reply:
@seriouslysean re: caching. I think the optimal way to manage caching the data, performance-wise, would be to update it incrementally as games are completed. We could store the leaderboard response data in its own table, update that when games and matches complete, and then the endpoints would just dump table data. 

I think that makes sense to look to as we continue to scale, but it's a larger lift so I think we hold off for the moment. Just adding the `select` clauses has given us a 10x performance boost and kept the endpoint's computational intensity well within the server's limits.